utilities/transactions/pessimistic_transaction.cc

```diff
@@ -559,7 +561,14 @@ Status PessimisticTransaction::TryLock(ColumnFamilyHandle* column_family,
   // any writes since this transaction's snapshot.
   // TODO(agiardullo): could optimize by supporting shared txn locks in the
   // future
-  if (skip_validate || snapshot_ == nullptr) {
+  if (!do_validate || snapshot_ == nullptr) {
+    assert(!assume_tracked || previously_locked);
+    if (assume_tracked) {
```

Comment:
Minor suggestion. Seems like this could be rewritten as:

  if (assume_tracked && !previous_locked) {
    s = Status::InvalidArgument();
  }